src/query.rs

```diff
@@ -0,0 +1,125 @@
+use std::any::type_name;
+
+use crate::{authentication::authenticate_user, order_item::OrderItem, user::User, Order};
+use async_graphql::{Context, Error, Object, Result};
+
+use bson::Uuid;
+use mongodb::{bson::doc, Collection, Database};
+use serde::Deserialize;
+
+/// Describes GraphQL order queries.
+pub struct Query;
+
+#[Object]
+impl Query {
+    /// Entity resolver for user of specific id.
+    #[graphql(entity)]
+    async fn user_entity_resolver<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(desc = "UUID of user to retrieve.")] id: Uuid,
+    ) -> Result<User> {
+        let db_client = ctx.data_unchecked::<Database>();
+        let collection: Collection<User> = db_client.collection::<User>("users");
+        query_object(&collection, id).await
+    }
+
+    /// Retrieves order of specific id.
+    async fn order<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(desc = "UUID of order to retrieve.")] id: Uuid,
+    ) -> Result<Order> {
+        let db_client = ctx.data_unchecked::<Database>();
+        let collection: Collection<Order> = db_client.collection::<Order>("orders");
+        let order = query_object(&collection, id).await?;
+        authenticate_user(&ctx, order.user._id)?;
+        Ok(order)
+    }
+
+    /// Entity resolver for order of specific id.
+    #[graphql(entity)]
+    async fn order_entity_resolver<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(key, desc = "UUID of order to retrieve.")] id: Uuid,
+    ) -> Result<Order> {
+        let db_client = ctx.data_unchecked::<Database>();
+        let collection: Collection<Order> = db_client.collection::<Order>("orders");
+        let order = query_object(&collection, id).await?;
+        authenticate_user(&ctx, order.user._id)?;
```

Comment:
entity resolvers do not need authentication in our authorization model
the side providing an entity initially is always responsible for doing authentication and authorization
the __entities query itself is not accessible via the Gateway